src/ecs/archetype/compact-archetype.ts

```diff
@@ -0,0 +1,34 @@
+/*MIT License
+
+© Copyright 2025 Adobe. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.*/
+import * as TABLE from "../../table/index.js";
+import { Archetype } from "./archetype.js";
+import { CoreComponents } from "../core-components.js";
+
+/**
+ * Compacts an archetype by reducing its rowCapacity to match rowCount.
+ * This shrinks the underlying buffers to remove unused capacity.
+ * Useful before serialization to avoid storing unused buffer space.
+ */
+export const compactArchetype = <C extends CoreComponents>(archetype: Archetype<C>): void => {
+    TABLE.compactTable(archetype);
+};
+
```

Comment:
We don't seem to need this as we could just call Table.compact(archetype) directly.